kayak_core/src/context.rs

```diff
@@ -267,222 +267,200 @@ impl KayakContext {
         self.widget_manager.calculate_layout();
     }
 
+    /// Processes the given input events
+    ///
+    /// Events are processed in three phases: Capture, Target, Propagate. These phases are based on their
+    /// associated [W3 specifications](https://www.w3.org/TR/uievents/#dom-event-architecture).
+    ///
+    /// ## Capture:
+    /// Currently, we do not support the Capture Phase. This is because the current event handling system is
+    ///   made to handle events as a single enum. To achieve proper capturing, widgets would need to be able to
+    ///   register separate event handlers so that specific ones could be captured while others would not. It
+    ///   should generally be okay to skip this as it's not a common use-case.
+    ///
+    /// ## Target:
+    ///   The Target Phase simply identifies the target for an event so that we can generate the propagation path
+    ///   for it.
+    ///
+    /// ## Propagate:
+    ///   The Propagate Phase (also known as the Bubble Phase) is where we bubble up the tree from the target node,
+    ///   firing the bubbled event along the way. At any point, the bubbling can be stopped by calling
+    ///   [`event.stop_propagation()`](Event::stop_propagation). Not every event can be propagated, in which case,
+    ///   they will only fire for their specified target.
```

Comment:
Can remove this from the documentation and just put it as a separate comment if that is preferred.